### PR TITLE
*: clean up the surrounding code of check pkg

### LINKED ISSUE #5105: Use `require.Eventually` to replace `testutil.WaitUntil`
## Enhancement Task

We can use [`require.Eventually`](https://pkg.go.dev/github.com/stretchr/testify@v1.7.1/require#Eventually) to replace the [`testutil.WaitUntil`](https://github.com/tikv/pd/blob/f964149e39f53e83b770641691ac6c3df0521e6d/pkg/testutil/testutil.go#L58) function we made ourselves.

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: close #5105, ref #4813.

### What is changed and how does it work?

```commit-message
Clean up the surrounding code of check pkg.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by disksing — APPROVED** [review]

File: pkg/testutil/testutil.go [hunk 1]
```diff
@@ -19,80 +19,48 @@ import (
...
 	"github.com/pingcap/kvproto/pkg/pdpb"
 	"github.com/stretchr/testify/require"
 	"google.golang.org/grpc"
 )
 
 const (
-	defaultWaitRetryTimes = 200
-	defaultSleepInterval  = time.Millisecond * 100
-	defaultWaitFor        = time.Second * 20
+	defaultWaitFor      = time.Second * 20
```
Comment (bufferflies) [inline pkg/testutil/testutil.go:28]: Why repeat it?  The similar code may be found in `/client/testutil.go`.

File: tests/client/client_test.go [hunk 2]
```diff
@@ -1306,7 +1306,7 @@ func (suite *clientTestSuite) TestScatterRegion() {
 		return resp.GetRegionId() == regionID &&
 			string(resp.GetDesc()) == "scatter-region" &&
 			resp.GetStatus() == pdpb.OperatorStatus_RUNNING
-	}, testutil.WithSleepInterval(1*time.Second))
+	}, testutil.WithTickInterval(1*time.Second))
```
Comment (bufferflies) [inline tests/client/client_test.go:1309]: How about using one style like `time.Second*10` or `10*time.Second`?

File: tests/server/tso/manager_test.go [hunk 3]
```diff
@@ -182,7 +182,7 @@ func TestNextLeaderKey(t *testing.T) {
 			cluster.CheckClusterDCLocation()
 			currName := cluster.WaitAllocatorLeader("dc-1")
 			return currName == name
-		}, testutil.WithSleepInterval(1*time.Second))
+		}, testutil.WithTickInterval(1*time.Second))
```
Comment (bufferflies) [inline tests/server/tso/manager_test.go:185]:
```suggestion
		}, testutil.WithTickInterval(time.Second))
```

File: pkg/testutil/testutil.go (hunk 1 above)
Comment (JmPotato) [reply pkg/testutil/testutil.go:28]: This is to prevent the client module from relying on the PD module.

**Pull request merged**